Automattic/pocket-casts-android
# Update categories pills UI

## Description
- Implements the action to update the category pills UI
- Moved `CategoriesListRowRedesignAdapter` to another file
- Added icons
- Created `CategoryPill` to wrap `DiscoverCategory`

This is part of the categories redesign. You can find references and figma here in our internal reference: `pdeCcb-4tv-p2`.

> [!note]
> The behavior to open categories view in the same discover view is not implemented yet. So clicking on some category may not provide the best experience

## Testing Instructions
1. `git apply` [categories-redesign.patch](https://github.com/Automattic/pocket-casts-android/files/14606106/categories-redesign.patch)
2. The categories should always be displayed in the following order: `True Crime, Comedy, Culture, History, Fiction, Technology`
3. Click on any category
4. The selected category must be updated to background `blue`, text color `white` ✅
5. The other categories must be hidden ✅
6. `All categories` button must be updated to X button ✅
7. Click on `X` button
8. The categories filter must be cleaned ✅
9. Click on `All categories` button
10. The modal with all categories must be opened with the arrow icon updated ✅

## Screenshots or Screencast 
[Screen_recording_20240314_143823.webm](https://github.com/Automattic/pocket-casts-android/assets/42220351/691c81c0-3695-47d5-8674-9003c2272927)


## Checklist
 
#### I have tested any UI changes...



cc: @david-gonzalez-a8c
